File: noop-java/src/main/java/com/exyui/android/debugbottle/components/DTEnv.java
package com.exyui.android.debugbottle.components;

import com.exyui.android.debugbottle.BuildConfig;

/**
 * Created by yuriel on 11/11/16.
 */

@SuppressWarnings("unused")
final public class DTEnv {
    public static final Env ENV = Env.RELEASE;
    public static final int VERSION_CODE = BuildConfig.VERSION_CODE;
    public static final String VERSION_NAME = BuildConfig.VERSION_NAME;

    public static Boolean isDebug() {
        return ENV == Env.DEBUG;
    }

    public static Boolean isRelease() {
        return ENV == Env.RELEASE;
    }

    public static Boolean isTest() {
        return ENV == Env.TEST;
    }

    public enum Env {
        DEBUG, RELEASE, TEST
    }
}

File: components/src/main/kotlin/com/exyui/android/debugbottle/components/DTEnv.kt
package com.exyui.android.debugbottle.components

/**
 * Created by yuriel on 11/11/16.
 */
@Suppress("unused")
object DTEnv {

    @JvmStatic val ENV = Env.DEBUG
    @JvmStatic val VERSION_CODE = BuildConfig.VERSION_CODE
    @JvmStatic val VERSION_NAME = BuildConfig.VERSION_NAME

    @JvmStatic fun isDebug() = ENV == Env.DEBUG
    @JvmStatic fun isRelease() = ENV == Env.RELEASE
    @JvmStatic fun isTest() = ENV == Env.TEST

    enum class Env {
        DEBUG, RELEASE, TEST
    }
}